File: webframework/src/main/java/com/brainz/framework/sample/db/repository/UserRepository.java
package com.brainz.framework.sample.db.repository;

import java.util.List;

import com.brainz.framework.sample.db.model.User;

import org.springframework.data.repository.CrudRepository;

public interface UserRepository extends CrudRepository<User, Long> {
	List<User> findByUserId(String userId);
	List<User> findAll();
}

File: src/main/kotlin/co/brainz/framework/sample/db/repository/UserRepository.kt
package co.brainz.framework.sample.db.repository;

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import co.brainz.framework.sample.db.model.User

@Repository
public interface UserRepository : CrudRepository<User, Long> {
    fun findByUserId(userId : String) : MutableList<User>
    override fun findAll() : MutableList<User> 
}
